hibernate4/src/main/java/com/fasterxml/jackson/datatype/hibernate4/HibernateProxySerializer.java

```diff
@@ -194,4 +200,40 @@ protected Object findProxied(HibernateProxy proxy)
         }
         return init.getImplementation();
     }
+    
+    // Alas, hibernate offers no public api to access this information, so we must resort to ugly hacks ...
```

Comment:
What to you mean by "expand on logic"? (How can I document the absence of a suitable API? Should I list what I tried, and why that didn't work?)